### PR TITLE
Add JAXopt recipe to conda-forge

### PR DESCRIPTION
**Pull request by dirmeier**
Hi all,

this PR adds [JAXopt](https://github.com/google/jaxopt) as recipe.

Greets,
Simon

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by dirmeier** [comment]
Hello @conda-forge/help-python, I think this is ready for review. Would you mind having a look? Thank, Simon

File: recipes/jaxopt/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+{% set version = "0.5" %}
+{% set sha256 = "9d48c30c78285ec5beea0264bb21fa69bbf9b82354136f0aaab0f956f82dd797" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz  
+  sha256: {{ sha256 }}
```
Comment (dopplershift) [inline recipes/jaxopt/meta.yaml:11]: Don't worry about templatizing the hash, just put it in here:
```suggestion
  sha256: 9d48c30c78285ec5beea0264bb21fa69bbf9b82354136f0aaab0f956f82dd797
```

File: recipes/jaxopt/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{% set name = "jaxopt" %}
+{% set version = "0.5" %}
+{% set sha256 = "9d48c30c78285ec5beea0264bb21fa69bbf9b82354136f0aaab0f956f82dd797" %}
```
Comment (dopplershift) [inline recipes/jaxopt/meta.yaml:3]:
```suggestion
```

File: recipes/jaxopt/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz  
+  sha256: {{ sha256 }}
+
+build:  
+  script: {{ PYTHON }} -m pip install . -vv  
+  skip: true  # [win or py<37]
```
Comment (dopplershift) [inline recipes/jaxopt/meta.yaml:15]: This looks like it's pure Python and can be noarch:
```suggestion
  noarch: python
```

We can ignore if this fails on Windows due to missing dependencies.

File: recipes/jaxopt/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  number: 0
+
+requirements:  
+  host:
+    - python
+    - setuptools
+    - pip
+  run:
+    - python
```
Comment (dopplershift) [inline recipes/jaxopt/meta.yaml:24]: For noarch:
```suggestion
    - python >=3.7
    - setuptools
    - pip
  run:
    - python >=3.7
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
Just some minor tweaks, including moving to noarch if possible.

**Comment by dirmeier** [comment]
Hi @dopplershift , thanks for the review. I commited your changes. The win build now fails as Jax doesn't have a windows install candidate.

**Review by dopplershift — APPROVED** [review]
Ignoring Windows CI failure due to missing dependency.

**Pull request merged**